lib/std/terminfo.zig

```diff
@@ -0,0 +1,1122 @@
+    .{ .short = "smgtb", .long = "set_tb_margin" },
+    .{ .short = "ehhlm", .long = "enter_horizontal_hl_mode" },
+    .{ .short = "elhlm", .long = "enter_left_hl_mode" },
+    .{ .short = "elohlm", .long = "enter_low_hl_mode" },
+    .{ .short = "erhlm", .long = "enter_right_hl_mode" },
+    .{ .short = "ethlm", .long = "enter_top_hl_mode" },
+    .{ .short = "evhlm", .long = "enter_vertical_hl_mode" },
+    .{ .short = "sgr1", .long = "set_a_attributes" },
+    .{ .short = "slength", .long = "set_pglen_inch" },
+};
+
+pub fn getCapsLookUpTable() !std.StringHashMap([]const u8) {
+    var data = std.StringHashMap([]const u8).init(std.heap.page_allocator);
+    for (dictionary) |term| 
+        _ = try data.put(term.short, term.long);
+    return data;
+}
+
+pub fn ReadLineIterator(comptime buffer_size: usize) type {
+    return struct {
+        const Self = @This();
+        const Stream = std.io.InStream(std.os.ReadError);
+        const FifoType = std.fifo.LinearFifo(u8, std.fifo.LinearFifoBufferType{ .Static = buffer_size });
+        const FilterCallback = fn([]u8)bool;
+        const MapCallback = fn([]u8)[]const u8;
+
+        fifo: FifoType,
+        file: std.fs.File,
+        stream: Stream,
+        filter_fn: FilterCallback,
+        map_fn: MapCallback,
+
+        pub fn init(file_path: []const u8) !Self {
+            const file = try std.fs.cwd().openFile(file_path, .{});
+            var in_stream = file.inStream();
+
+            return Self {
+                .fifo = FifoType.init(),
+                .file = file,
+                .stream = in_stream.stream,
+                .filter_fn = struct { fn filter(line: []u8) bool { return true; } }.filter,
+                .map_fn = struct { fn map(line: []u8) []u8 { return line; } }.map,
+            };
+        }
+
+        pub fn deinit(self: *Self) void {
+            self.file.close();
+        }
+
+        pub fn setFilter(self: *Self, filter_fn: FilterCallback) void {
+            self.filter_fn = filter_fn;
+        }
+
+        pub fn setMap(self: *Self, map_fn: MapCallback) void {
+            self.map_fn = map_fn;
+        }
+
+        pub fn next(self: *Self) !?[]const u8 {
+            while (try (&self.stream).readUntilDelimiterOrEof(self.fifo.buf[0..], '\n')) |line| {
+                if (self.filter_fn(line)) return self.map_fn(line);
+            }
+            return null;
+        }
+
+        pub fn collect(self: *Self) !std.Buffer {
+            var buffer = try std.Buffer.initCapacity(std.heap.page_allocator, buffer_size);
+
+            while (try self.next()) |line| {
+                try buffer.append(line);
+            }
+
+            return buffer;
+        }
+    };
+}
+
+pub const Terminfo = struct {
+    names: std.StringHashMap(bool),
+    caps: std.StringHashMap([]const u8),
+
+    pub fn init(terminfo_path: []const u8, cap_delimiter: []const u8) !Terminfo {
+        var names = std.StringHashMap(bool).init(std.heap.page_allocator);
```

Comment:
never assume an allocator (take one as a parameter).
Also save the allocator in the struct and use it from `deinit`.